frontend/src/modules/app/ToolBar.tsx

```diff
@@ -243,76 +229,73 @@ export const ToolBar = ({
       case VentilationMode.niv_pc:
       case VentilationMode.niv_ps:
       case VentilationMode.psv:
-        dispatch(
-          commitRequest<ParametersRequest>(MessageType.ParametersRequest, {
-            peep: parameterRequestDraft.peep,
-            vt: parameterRequestDraft.vt,
-            rr: parameterRequestDraft.rr,
-            fio2: parameterRequestDraft.fio2,
-          }),
-        );
+        dispatchParameterRequest({
+          peep: parameterRequestDraft.peep,
+          vt: parameterRequestDraft.vt,
+          rr: parameterRequestDraft.rr,
+          fio2: parameterRequestDraft.fio2,
+        });
         break;
       default:
         break;
     }
-  }, [isVentilatorOn, parameterRequestDraft, alarmLimitsRequestDraft, currentMode, dispatch]);
+  };
 
   /**
-   * Disabled Start/Pause Ventilation button when backend connection is lost
+   * Set Label for Start/Pause Ventilation button
    */
   useEffect(() => {
-    if (storeReady) {
-      setLandingLabel('Start');
-      setIsDisabled(false);
-      setLabel(ventilating ? 'Pause Ventilation' : 'Start Ventilation');
-    } else if (ventilatingStatus) {
-      setIsDisabled(true);
+    if (ventilatingStatus) {
       setLabel('Connecting...');
     } else {
-      setLandingLabel('Loading...');
-      setIsDisabled(true);
-      setLabel('Loading...');
+      setLabel(ventilating ? 'Pause Ventilation' : 'Start Ventilation');
     }
   }, [storeReady, ventilatingStatus, ventilating]);
 
-  useEffect(() => {
-    if (ventilating) {
-      return;
-    }
-
-    initParameterUpdate();
-  }, [ventilating, initParameterUpdate]);
-
   /**
-   * Update Label on Button based on ventilation status
+   * Switch to dashboard page if ventilating
    */
   useEffect(() => {
-    if (ventilating) {
+    if (storeReady && ventilating) {
       history.push(DASHBOARD_ROUTE.path);
```

Comment:
Is this the place which takes the frontend to the dashboard once ventilation starts? If so, perhaps this is where we could take the frontend to the quickstart screen once ventilation is paused.

Reply:
This `useEffect` could be a candidate for decomposing out to a separate (very small) component which takes care of all auto-navigation, which we would include in the main layout; its behavior is independent of the toolbar so I'm not sure the code needs to live in this file. This would be for a future refactoring PR.

Reply:
> Is this the place which takes the frontend to the dashboard once ventilation starts? If so, perhaps this is where we could take the frontend to the quickstart screen once ventilation is paused.

Yes this is the useEffect that does that. yep most likely